docs/dev_guide.md

```diff
@@ -1,31 +1,166 @@
 # Dev Guide
 
-## Adding a new param in GCSFuse
+## Adding new features to GCSFuse
+
+**If you're interested in adding a new feature, please open a GitHub issue to
+discuss your proposal with the GCSFuse team first before making the
+contribution.**
+
+Rollout of any new feature should be done in 2 phases:
+
+**1. Experimental Phase:**
```

Comment:
Where will we keep/document features in experimental phase and their flags? I dont want to document them on public docs, so can we have a doc under contributing that shows all contributed features and their status? And ask users to update that as well?

Reply:
IMO, this yaml file (which we use to generate flag/config code) should serve the purpose of documentation too: https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cfg/params.yaml

"experimental" prefix in the `flag-name:` and `hide-flag:` parameter should convey the status.